### ISSUE TITLE
[BUG] Migration error when migrating to 1.5.1 prevents Formbricks from starting (20240204084813_remove_creator_cascade_delete_in_survey)

### ISSUE THREAD

**Issue by agrudsky** [formbricks/formbricks#2115]
### Issue Summary

I am running Formbricks 1.5.0 using Docker Compose. When I attempt to upgrade to 1.5.1, I receive the following error:

```
formbricks-1  | 64 migrations found in prisma/migrations
formbricks-1  | 
formbricks-1  | Applying migration `20240204084813_remove_creator_cascade_delete_in_survey`
postgres-1    | 2024-02-22 03:37:13.023 UTC [29] ERROR:  constraint "Survey_createdBy_fkey" of relation "Survey" does not exist
postgres-1    | 2024-02-22 03:37:13.023 UTC [29] STATEMENT:  -- DropForeignKey
postgres-1    |         ALTER TABLE "Survey" DROP CONSTRAINT "Survey_createdBy_fkey";
postgres-1    | 
postgres-1    |         -- AddForeignKey
postgres-1    |         ALTER TABLE "Survey" ADD CONSTRAINT "Survey_createdBy_fkey" FOREIGN KEY ("createdBy") REFERENCES "User"("id") ON DELETE SET NULL ON UPDATE CASCADE;
postgres-1    | 
formbricks-1  | Error: P3018
formbricks-1  | 
formbricks-1  | A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
formbricks-1  | 
formbricks-1  | Migration name: 20240204084813_remove_creator_cascade_delete_in_survey
formbricks-1  | 
formbricks-1  | Database error code: 42704
formbricks-1  | 
formbricks-1  | Database error:
formbricks-1  | ERROR: constraint "Survey_createdBy_fkey" of relation "Survey" does not exist
formbricks-1  | 
formbricks-1  | DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E42704), message: "constraint \"Survey_createdBy_fkey\" of relation \"Survey\" does not exist", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("tablecmds.c"), line: Some(12035), routine: Some("ATExecDropConstraint") }
formbricks-1  | 
formbricks-1  | 
formbricks-1 exited with code 0
```

This results in the Formbricks container in a restart loop: 

```
formbricks-1  | migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
formbricks-1  | The `20240204084813_remove_creator_cascade_delete_in_survey` migration started at 2024-02-22 03:37:13.014207 UTC failed
```

To get Formbricks to boot again, I had to revert to the 1.5.0 image in my Compose file, and run this command before starting the Compose instance again:
`sudo docker compose run formbricks pnpm dlx prisma migrate resolve --rolled-back "20240204084813_remove_creator_cascade_delete_in_survey"`

### Steps to Reproduce

1. `docker compose down`
2. `docker compose pull` to get latest Formbricks image (1.5.1)
3. `docker compose up -d`
4. Error is thrown and Formbricks container does not start

### Expected behavior

Formbricks should start without migration issues

### Other information

After reverting back to 1.5.0, and rolling back the migration as above, Formbricks works fine.

### Environment

- [ ] Formbricks Cloud (app.formbricks.com)
- [X] Self-hosted Formbricks

### Desktop (please complete the following information)

```markdown
- OS: Ubuntu 20.04.6 LTS
- Node: using Formbricks 1.5.1 Docker image
- npm: using Formbricks 1.5.1 Docker image
```

**Comment by agrudsky** [comment]
Looks like this was due to database corruption. I was able to fix this with:

1. Take a database backup dump: `sudo docker compose exec -it postgres pg_dump -U postgres -d formbricks  > ./formbricks.sql`
2. Drop the database: `sudo docker compose exec -it postgres dropdb -U formbricks formbricks`
3. Create a blank database: `sudo docker compose exec -it postgres createdb -U formbricks formbricks`
4. Restore from backup: `sudo docker compose exec -it postgres psql -U postgres -d formbricks -f formbricks.sql`

After this, I was able to migrate to 1.6.0 (and follow its migration guide) successfully.
